persistence/persistence-jdbc/src/test/java/uk/gov/justice/services/jdbc/persistence/PreparedStatementWrapperTest.java

```diff
@@ -0,0 +1,180 @@
+        final String query = "someQuery3";
+        when(connection.prepareStatement(query)).thenReturn(preparedStatement);
+        when(preparedStatement.executeQuery()).thenReturn(resultSet);
+
+
+        final PreparedStatementWrapper ps = PreparedStatementWrapper.valueOf(connection, query);
+        ps.executeQuery();
+
+        ps.close();
+        verify(connection).close();
+        verify(preparedStatement).close();
+        verify(resultSet).close();
+    }
+
+    @Test
+    public void shouldDelegateSetObjectMethodCall() throws SQLException {
+        final String query = "dummy";
+        when(connection.prepareStatement(query)).thenReturn(preparedStatement);
+        final PreparedStatementWrapper ps = PreparedStatementWrapper.valueOf(connection, query);
+        final int parameterIndex = 1;
+        final Object obj = new Object();
+        ps.setObject(parameterIndex, obj);
+
+        verify(preparedStatement).setObject(parameterIndex, obj);
+    }
+
+    @Test
+    public void shouldDelegateSetStringMethodCall() throws SQLException {
+        final String query = "dummy";
+        when(connection.prepareStatement(query)).thenReturn(preparedStatement);
+        final PreparedStatementWrapper ps = PreparedStatementWrapper.valueOf(connection, query);
+        final int parameterIndex = 2;
+        final String str = "aaa";
+        ps.setString(parameterIndex, str);
+
+        verify(preparedStatement).setString(parameterIndex, str);
+    }
+
+    @Test
+    public void shouldDelegateSetLongMethodCall() throws SQLException {
+        final String query = "dummy";
+        when(connection.prepareStatement(query)).thenReturn(preparedStatement);
+        final PreparedStatementWrapper ps = PreparedStatementWrapper.valueOf(connection, query);
+        final int parameterIndex = 2;
+        final Long lng = 123l;
+        ps.setLong(parameterIndex, lng);
+
+        verify(preparedStatement).setLong(parameterIndex, lng);
+    }
+
+    @Test
+    public void shouldDelegateExecuteUpdateMethodCall() throws SQLException {
+        final String query = "someQuery3";
+        when(connection.prepareStatement(query)).thenReturn(preparedStatement);
+        when(preparedStatement.executeUpdate()).thenReturn(4);
+
+
+        final PreparedStatementWrapper ps = PreparedStatementWrapper.valueOf(connection, query);
+        assertThat(ps.executeUpdate(), is(4));
+    }
+
+    @Test
+    public void shouldCloseStatementOnExceptionOnSetObject() throws SQLException {
+        final String query = "dummy";
+        when(connection.prepareStatement(query)).thenReturn(preparedStatement);
+        final PreparedStatementWrapper ps = PreparedStatementWrapper.valueOf(connection, query);
+        final int parameterIndex = 1;
+        final Object obj = new Object();
+        doThrow(new SQLException()).when(preparedStatement).setObject(parameterIndex, obj);
+        try {
+            ps.setObject(parameterIndex, obj);
```

Comment:
That is not a subject of this test. 
It's called "shouldCloseStatementOnExceptionOnSetObject" and if it fails on assertFail that would suggest that the connections are not closed on exception
If you think it's valuable to test if the exception is rethrown then that would require a separate test case.